### ISSUE TITLE
Add rule for `prefer-enum-initializers`

### ISSUE THREAD

**Issue by francoismora** [SonarSource/SonarJS#3809]
https://typescript-eslint.io/rules/prefer-enum-initializers

 - Covered?
   - Not covered
 - Makes sense?
   - Require each enum member value to be explicitly initialized
 - No options, no quickfix
 - Decorate or implement it ourselves. We should raise an issue if and only if there are some enum values initialized but not all like:

```typescript
enum Status {
 Open = 1,
 Close,
}
```
